WebServices/AzureNotificationHub/NotificationHubSample/NotificationHubSample.iOS/AppDelegate.cs

```diff
@@ -60,7 +60,7 @@ public override void RegisteredForRemoteNotifications(UIApplication application,
             Hub = new SBNotificationHub(AppConstants.ListenConnectionString, AppConstants.NotificationHubName);
 
             // update registration with Azure Notification Hub
-            Hub.UnregisterAllAsync(deviceToken, (error) =>
+            Hub.UnregisterAll(deviceToken, (error) =>
```

Comment:
Note: it is also possible to use the `UnregisterAllAsync` method but the signature has changed so that it doesn't have a callback. In order to guarantee that these methods were executed sequentially you'd have to do something like:
```csharp
try {
    await Hub.UnregisterAllAsync(deviceToken);
}
catch (Exception e) {
    // do something with the error
}
// execute the next method
```

Using the `await` keyword would cause warnings since it's not used in an Async suffixed method. Finally, this method is not fired from a UI thread so there's no real reason to use the async methods. I felt like it was more clear to simply use the non-async methods and keep the code pattern recommended in multiple places in both Xamarin and Azure docs.

Reply:
The changes in the `async` method signature were caused by changes to how we generate bindings. Our bindings now adhere more strictly to the classic async/await pattern and no longer have the concept of error callbacks.